exporter/awsemfexporter/metric_declaration.go

```diff
@@ -136,6 +176,44 @@ func (m *MetricDeclaration) ExtractDimensions(labels map[string]string) (dimensi
 	return
 }
 
+// Initialize LabelMatcher with default values and compile regex string.
+func (lm *LabelMatcher) Init() (err error) {
+	// Throw error if no label names are specified
+	if len(lm.LabelNames) == 0 {
+		return errors.New("label matcher must have at least one label name specified")
+	}
+	if len(lm.Separator) == 0 {
+		lm.Separator = ";"
+	}
+	if len(lm.Regex) == 0 {
+		lm.Regex = ".+"
+	}
+	lm.compiledRegex = regexp.MustCompile(lm.Regex)
+	return
+}
+
+// Returns true if given set of labels matches the LabelMatcher's rules.
+func (lm *LabelMatcher) Matches(labels map[string]string) bool {
+	concatenatedLabels := lm.getConcatenatedLabels(labels)
+	return lm.compiledRegex.MatchString(concatenatedLabels)
+}
+
+// Concatenate label values of matched labels using separator defined by the LabelMatcher's rules.
+func (lm *LabelMatcher) getConcatenatedLabels(labels map[string]string) string {
+	buf := new(bytes.Buffer)
+	isFirstLabel := true
+	for _, labelName := range lm.LabelNames {
```

Comment:
We shouldn't concatenate all the labels, we need to concatenate the labels which is configured and honor the order in configuration.